src/sql/magic.py

```diff
@@ -619,6 +622,73 @@ def _persist_dataframe(
         display.message_success(f"Success! Persisted {table_name} to the database.")
 
 
+def find_toml_path():
+    """
+    Recursively finds an absolute path to pyproject.toml starting
+    from current directory
+    """
```

Comment:
This can be a generic function `find_path_from_root` that accepts filename as parameter

Reply:
The function is now `find_path_from_root` in `util.py`.